Maya: Small Tweaks on Validator for Look Default Shader Connection for Maya 2024

moonyuet opened the pull request:
## Changelog Description
Resolve https://github.com/ynput/OpenPype/issues/5269

## Additional info
Please test Maya 2024 and any version elder than Maya 2024

## Testing notes:
1. Create Look with or without tx options.
2. Publish
3. It should run smoothly

BigRoy commented:
Actually - it's totally valid to still have that connection Maya 2024+ and I believe the issue _may_ also exist in older versions of Maya too since I believe for some time you were already able to toggle which is the default shader Maya should create in new scenes.

Note that this would only apply to **new** scenes - e.g. loading an old scene in Maya 2024 won't show the issue usually either.

We fixed it like [this](https://github.com/BigRoy/OpenPype/blob/colorbleed/openpype/hosts/maya/plugins/publish/validate_look_default_shaders_connections.py) so that we allow the connection to be **either** the standardSurface default connection **or** the lambert1 default connection, supporting both.

We also added a repair action ❤️  :)

moonyuet commented:
> Actually - it's totally valid to still have that connection Maya 2024+ and I believe the issue _may_ also exist in older versions of Maya too since I believe for some time you were already able to toggle which is the default shader Maya should create in new scenes. In Maya 2024 they however changed the default shader by default to `standardSurface` but you can still toggle the default to lambert1 as well.
> 
> Note that this would only apply to **new** scenes - e.g. loading an old scene in Maya 2024 won't show the issue usually either. Since the default is only applied to new scenes.
> 
> We fixed it like [this](https://github.com/BigRoy/OpenPype/blob/colorbleed/openpype/hosts/maya/plugins/publish/validate_look_default_shaders_connections.py) so that we allow the connection to be **either** the standardSurface default connection **or** the lambert1 default connection, supporting both.
> 
> We also added a repair action ❤️ :)
> 
> It might be worth just copying that logic from our codebase?

I tested with the code and compared the logic with mine. I agree that the connection issue happened in other versions and we need to come up with the best solutions for validating the files which is created by the old version of maya and opening in maya 2024. So yeah, I think it's more stable to have the code tweaked as if what you did in the branch. Thank you for helping out! ❤️
